app/models/dish.rb

```diff
@@ -0,0 +1,22 @@
+class Dish < ApplicationRecord
+  # scope by_name order(:type )
+  has_many :ordered_soups, class_name: Order.name, foreign_key: 'soup_id'
+  has_many :ordered_main_dishes, class_name: Order.name, foreign_key: 'main_dish_id'
+  has_many :ordered_beverages, class_name: Order.name, foreign_key: 'beverage_id'
+  has_and_belongs_to_many :menu_days
+
+  validate :price, presicion
+
+
+  enum dish_type: {
+    soup: SOUP = 'soup',
+    main_dish: MAIN_DISH = 'main',
+    beverage: BEVERAGE = 'beverage'
+  }
+
+  scope :soups,-> { where(dish_type: :soup) }
```

Comment:
Nope. You should not do this. Because you have enum and you have such scopes automatically. For new developers this will be confused. you have scopes by enum type. Dish.soup, Dish.main_dish, Dish.beverage.It is ok when it is not in plural when it is enum